internal/db/schema/migrations/oss/postgres/100/01_iam_role_global.up.sql

```diff
@@ -0,0 +1,163 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+
+  create table iam_role_global_grant_scope_enm (
+    name text not null primary key
+      constraint only_predefined_scope_types_allowed
+        check(
+          name in (
+            'descendants',
+            'children',
+            'individual'
+          )
+        )
+  );
+
+  insert into iam_role_global_grant_scope_enm (name)
+  values
+    ('descendants'),
+    ('children'),
+    ('individual');
+
+  create or replace function insert_role_subtype() returns trigger
+  as $$
+  begin
+    insert into iam_role
+      (public_id, scope_id)
+    values
+      (new.public_id, new.scope_id);
+    return new;
+  end;
+  $$ language plpgsql;
+
+  create or replace function insert_grant_scope_update_time() returns trigger
+  as $$
+  begin
+    if new.grant_scope is distinct from old.grant_scope then
+      new.grant_scope_update_time = now();
+    end if;
+    return new;
+  end;
+  $$ language plpgsql;
+
+  create or replace function insert_grant_this_role_scope_update_time() returns trigger
+  as $$
+  begin
+    if new.grant_this_role_scope is distinct from old.grant_this_role_scope then
+      new.grant_scope_update_time = now();
+    end if;
+    return new;
+  end;
+  $$ language plpgsql;
+
+  -- Add trigger to update the new column on every iam_role subtype update.
+  -- This is used to update the update_time of the iam_role table
+  -- when either the name or the description of the subtype tables are updated.
+  -- This is only applicable to the name and description columns because we
+  -- do not want the update_time to be updated when the grant_scope or grant_this_role_scope
+  -- columns are updated.
+  create function update_iam_role_table_update_time() returns trigger
+  as $$
+  begin
+    if (new.name is distinct from old.name) or (new.description is distinct from old.description) then
+      update iam_role set update_time = now() where public_id = new.public_id;
+      return new;
+    end if;  
+  end;
+  $$ language plpgsql;
+  comment on function update_iam_role_table_update_time() is
+    'update_iam_role_table_update_time is used to automatically update the update_time '
+    'of the base table whenever one of the subtype tables are updated';
+
+  -- global iam_role must have a scope_id of global
+  create table iam_role_global (
+    public_id wt_role_id not null primary key
```

Comment:
```suggestion
    public_id wt_role_id primary key
```
It is not necessary to declare a primary key as `not null` in PostgreSQL